t/lib/WTSI/NPG/HTS/PacBio/Sequel/RunPublisherTest.pm

```diff
@@ -860,9 +870,9 @@ sub check_primary_metadata {
     if ($obj->find_in_metadata($TARGET)){
       my @tag_meta = $obj->find_in_metadata($TAG_SEQUENCE);
       my @tags = map {$_->{value}} @tag_meta;
-      $expected_id = $product->generate_product_id($run_name, $well_label, \@tags);
+      $expected_id = $product->generate_product_id($run_name, $well_label, \@tags, $pn);
     }else{
-      $expected_id = $product->generate_product_id($run_name, $well_label);
+      $expected_id = $product->generate_product_id($run_name, $well_label, undef, $pn);
```

Comment:
One final thing that I wonder if we can improve is the case where the tags are empy, but there is a plate name. Does this occur in practice?
Having a list of positional args with an `undef` in the middle just so that the additonal positional param can be added is a bit ugly and prone to mistakes when used.

One option is to wrap `generate_product_id` to allow the plate number to be supplied as a named argument, like here https://github.com/wtsi-npg/npg_irods/blob/devel/lib/WTSI/NPG/HTS/Illumina/Annotator.pm#L305